### PR TITLE
Localise cleanup

### PR DESCRIPTION
**Pull request by danishjafri88**
## Pull Request Checklist

- [ ] My patch has gone through review and I have addressed review comments
- [ ] My patch has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!`

- [ ] I have updated the *Unit Tests* to cover new or changed functionality
- [ ] I have updated the *UI Tests* to cover new or changed functionality
- [ ] I have marked the bug with `[needsuplift]`
- [ ] I have made sure that localizable strings use `NSLocalizableString()`

## Screenshots

If your patch includes user interface changes that you would like to suggest or that you would like UX to look at, please include them here.

## Notes for testing this patch

If useful, please leave notes for QA, explaining what this patch changes and how it can be best tested and verified.

## Additional Info
The Duplicates.txt file contains list of duplicate strings with different contexts. Though similar contextual duplicates have been removed, please review and mark in the remaining as well. Strings are mostly in BraveStrings.swift and Strings.swift. Some in BraveStrings are marked with their respective usage swift file as PRAGMA MARK. We can either let it be in BraveStrings or move to the respective files. Scripts are stored in the l10n folder in root dir.

### CODE REVIEW

**Comment by kylehickinson** [comment]
Please rebase this on `development@HEAD`, Travis shows there are build errors for missing Strings:

```
[21:37:30]: ▸ ❌  /Users/travis/build/brave/brave-ios/Client/Frontend/Browser/Search/SearchSuggestionsPromptView.swift:39:28: type 'Strings' has no member 'Turn_on_search_suggestions'
[21:37:30]: ▸ promptLabel.text = Strings.Turn_on_search_suggestions
[21:37:30]: ▸                                                   ^
[21:37:30]: ▸ ❌  /Users/travis/build/brave/brave-ios/Client/Frontend/Browser/Search/SearchSuggestionsPromptView.swift:46:34: type 'Strings' has no member 'Yes'
[21:37:30]: ▸ promptYesButton.setTitle(Strings.Yes, for: .normal)
[21:37:30]: ▸                            ^~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~
[21:37:30]: ▸ ❌  /Users/travis/build/brave/brave-ios/Client/Frontend/Browser/Search/SearchSuggestionsPromptView.swift:57:33: type 'Strings' has no member 'No'
[21:37:30]: ▸ promptNoButton.setTitle(Strings.No, for: .normal)
[21:37:30]: ▸                                  ^~~~~~~ ~~~
[21:37:30]: ▸ ❌  /Users/travis/build/brave/brave-ios/Client/Frontend/Browser/Search/SearchSuggestionsPromptView.swift:39:28: type 'Strings' has no member 'Turn_on_search_suggestions'
[21:37:30]: ▸ promptLabel.text = Strings.Turn_on_search_suggestions
[21:37:30]: ▸                                            ^~~
[21:37:30]: ▸ ❌  /Users/travis/build/brave/brave-ios/Client/Frontend/Browser/Search/SearchSuggestionsPromptView.swift:46:34: type 'Strings' has no member 'Yes'
[21:37:30]: ▸ promptYesButton.setTitle(Strings.Yes, for: .normal)
[21:37:30]: ▸                            ^~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~
[21:37:30]: ▸ ❌  /Users/travis/build/brave/brave-ios/Client/Frontend/Browser/Search/SearchSuggestionsPromptView.swift:57:33: type 'Strings' has no member 'No'
[21:37:30]: ▸ promptNoButton.setTitle(Strings.No, for: .normal)
```

File: BraveShared/BraveStrings.swift [hunk 1]
```diff
@@ -15,344 +15,391 @@
 
 import Shared
 
-// SendTo extension.
+// PRAGMA MARK: Common Strings Here.
```
Comment (iccub) [inline BraveShared/BraveStrings.swift]: Just `// MARK: xxx` can be used, or `// MARK: - xxx` if you want section to be underlined. Replace it in all places

File: BraveShared/BraveStrings.swift [hunk 2]
```diff
@@ -15,344 +15,391 @@
...
+// PRAGMA MARK: BasePasscodeViewController.swift
 public extension Strings {
-    public static let ClearPrivateData = NSLocalizedString("Clear Private Data", comment: "Button in settings that clears private data for the selected items. Also used as section title in settings panel")
+    public static let PasscodeConfirmMisMatchErrorText = NSLocalizedString("PasscodeConfirmMisMatchErrorText", tableName: "BraveShared", value: "Passcodes didn’t match. Try again.", comment: "Error message displayed to user when their confirming passcode doesn't match the first code.")
+    public static let PasscodeMatchOldErrorText = NSLocalizedString("PasscodeMatchOldErrorText", tableName: "BraveShared", value: "New passcode must be different than existing code.", comment: "Error message displayed when user tries to enter the same passcode as their existing code when changing it.")
 }
 
-// Error pages.
+// PRAGMA MARK: AuthenticationManagerConstants.swift
 public extension Strings {
```
Comment (iccub) [inline BraveShared/BraveStrings.swift]: Why is it empty?

File: l10n/LocalizedMatcher.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+egrep --exclude-dir={ThirdParty,Carthage,fastlane,L10nSnapshotTests,l10n} --include=\*.swift -nR "NSLocalizedString(.*[\n]*)" . > localizedStringLocations.txt
```
Comment (iccub) [inline l10n/LocalizedMatcher.sh:2]: Could you add some comment in the script about what it does?

**Review by iccub — CHANGES_REQUESTED** [review]

**Comment by iccub** [comment]
Also is there any reason for us having `Strings.swift` and `BraveStrings.swift`  separately?

**Comment by jhreis** [comment]
@iccub in reference to last question about `strings` vs `bravestrings`.

Our initial hope was to leave `strings.swift` as untouched as possible since it is part of the FF `Shared` module. However, I suspect this won't work given how much we needed to remove.

That being said, we can re-address later, but would be part of a larger string refactor at another time.

File: BraveShared/BraveStrings.swift (hunk 1 above)
Comment (danishjafri88) [reply BraveShared/BraveStrings.swift]: Done :)

File: BraveShared/BraveStrings.swift (hunk 2 above)
Comment (danishjafri88) [reply BraveShared/BraveStrings.swift]: Duplicated entry cause os a script. Removed

File: l10n/LocalizedMatcher.sh (hunk 3 above)
Comment (danishjafri88) [reply l10n/LocalizedMatcher.sh:2]: Added a description.

**Review by iccub — APPROVED** [review]

**Pull request merged**